### PR TITLE
chore(node): bumping Superset to Node 18

### PR DESCRIPTION
**Pull request by rusackas**
### SUMMARY

Not sure all the actions bumps are needed, but I'll just aim high and hope we can get everything updated.

This does NOT bump the websocket app or docs to Node 18.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by mistercrunch — APPROVED** [review]
LGTM after hold is removed

**Comment by rusackas** [comment]
@michael-s-molina any objection to merging this now? We have these warnings cropping up, which makes me almost tempted to call this a "fix" rather than a "chore". It's non-blocking for 4.0 and doesn't _need_ to be cherried, necessarily. But I also don't want to rock the boat on your release :) 

image

**Comment by michael-s-molina** [comment]
My only concern is this causing bugs on master which will be reported as issues and we'll need to check if they happen on 4.0. If you think the probability of new bugs is low then go ahead and merge it.

**Comment by rusackas** [comment]
All conflicts resolved... hopefully we can merge this soon :D

File: .github/workflows/tech-debt.yml [hunk 1]
```diff
@@ -32,8 +32,8 @@ jobs:
       - name: Set up Node.js
         uses: actions/setup-node@v4
         with:
-          node-version: "16"
-
+          node-version: '18'
+
```
Comment (rusackas) [inline .github/workflows/tech-debt.yml:36]:
```suggestion
```suggestion
          node-version: '18'

```
```

**Comment by mistercrunch** [comment]
cleaning

**Comment by eschutho** [comment]
@rusackas It looks like npm 9+ use lockfile version 3, so I also updated that check here: 
https://github.com/apache/superset/pull/29723/files

**Pull request merged**